lib/spack/spack/schema/packages.py

```diff
@@ -69,6 +69,20 @@ def deprecate_paths_and_modules(instance, deprecated_properties):
                 'default': {},
                 'additionalProperties': False,
                 'properties': {
+                    'require': {
+                        'oneOf': [
+                            # A condition group: at least one member is applied
+                            {'type': 'array',
+                             'items': {
+                                 'type': 'object',
+                                 'properties': {
+                                     'one_of': {'type': 'array'}
```

Comment:
Note for reviewer: YAML schema has a concept of "oneOf" (to specify that a YAML element can have multiple types). This particular schema includes a notion of an element which is itself called "one_of", which appears in the literal YAML as:

```
packages:
  openmpi:
    require:
    - one_of: [%gcc, %clang]   # or these can be vertical
    - one_of: [@4.0.1, @4.0.2]
```

In this schema, you can either have this or something like

```
packages:
  openmpi:
    require: "%gcc" # Note this is not expressing the same thing as the previous example
```

So "require" can either be a list of "one_of" elements, or a string.

Reply:
Am I correct that this schema still needs to be specified, at least for the `array` part? Also, I tried:
```yaml
packages:
  zlib:
    require: '@1.2.12'
```
and got the following parsing error:
```console
$ spack  spec libpng
==> Error: Invalid version specifier  Encountered when parsing spec:
    @
     ^
```